app/src/components/MessageList.js

```diff
@@ -0,0 +1,201 @@
+                SNAPSHOT_INSTRUMENT,
+                lowerSodOffset.toString(),
+                nMessages,
+            );
+
+            const { messages, pageInfo } = reverseMessagesResponse.data;
+
+            this.setState({
+                messages,
+                lastSodOffsetTop: lowerSodOffset,
+                lastSodOffsetBottom: BigInt(pageInfo.sod_offset),
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    };
+
+    /**
+     * @desc Paging handler for upwards and downwards hitting of the multidirectional scroll
+     * @param direction
+     */
+    handleHitEdge(direction) {
+        const { lastSodOffsetTop, lastSodOffsetBottom } = this.state;
+        // eslint-disable-next-line react/destructuring-assignment
+        const existingMessages = this.state.messages;
+
+        if (direction === 'top') {
+            const nMessages = -MESSAGE_LIST_DEFAULT_PAGE_SIZE;
+            OrderBookService.getMessageList(SNAPSHOT_INSTRUMENT, lastSodOffsetTop.toString(), nMessages)
+                .then(response => {
+                    const { pageInfo, messages } = response.data;
+                    this.setState({
+                        messages: messages ? messages.concat(existingMessages) : existingMessages,
+                        lastSodOffsetTop: BigInt(pageInfo.sod_offset),
+                    });
+                })
+                .catch(err => {
+                    console.log(err);
+                });
+        } else if (direction === 'bottom') {
+            OrderBookService.getMessageList(SNAPSHOT_INSTRUMENT, lastSodOffsetBottom.toString())
+                .then(response => {
+                    const { pageInfo, messages } = response.data;
+                    this.setState({
+                        messages: messages ? existingMessages.concat(messages) : existingMessages,
+                        lastSodOffsetBottom: BigInt(pageInfo.sod_offset),
+                    });
+                })
+                .catch(err => {
+                    console.log(err);
+                });
+        }
+    }
+
+    /**
+     * @desc Renders every row of the message list table. every row corresponds to a message
+     * @returns {*}
+     */
+    renderTableData() {
+        const { classes, lastSodOffset } = this.props;
+        const { messages } = this.state;
+        return messages.map(message => {
+            const {
+                timestamp, message_type, order_id, share_qty, price, direction, sod_offset,
+            } = message;
+
+            const { timeNanoseconds, dateNanoseconds } = splitNanosecondEpochTimestamp(timestamp);
+            const date = epochToDateString(dateNanoseconds);
+            const time = nanosecondsToString(timeNanoseconds);
+
+            return (
+                <Box
+                    key={`${sod_offset} ${timestamp}`}
+                    className={lastSodOffset.toString() === sod_offset
+                        ? classes.currentMessageRow
```

Comment:
Nice! From what I understand, this highlights the last APPLIED message, right?

Reply:
yes, the "correct message with respect to current orderbook state" -issue #54